### PR TITLE
Support SPMD fsdp compute dtype

### PR DESCRIPTION
**Pull request by lausannel**

### CODE REVIEW

File: torch_xla/experimental/spmd_fully_sharded_data_parallel.py [hunk 1]
```diff
@@ -192,3 +209,49 @@ def _auto_wrap(
             "if using an `auto_wrap_policy`")
 
     recursive_wrap(**auto_wrap_kwargs, **fsdp_kwargs)
+
+
+def _cast_floats_tensors(dtype: torch.dtype, *args: Any,
```
Comment (baoleai) [inline torch_xla/experimental/spmd_fully_sharded_data_parallel.py]: can we directly import from xla_fully_sharded_data_parallel.py?

File: torch_xla/experimental/spmd_fully_sharded_data_parallel.py [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
-import numpy as np
-
 import torch_xla
 import torch_xla.core.xla_model as xm
 import torch_xla.distributed.spmd as spmd
-from torch_xla.distributed.fsdp.wrap import recursive_wrap
 from torch_xla.distributed.fsdp._init_utils import _materialize_module
+from torch_xla.distributed.fsdp.wrap import recursive_wrap
+from torch_xla.distributed.fsdp.xla_fully_sharded_data_parallel import (
+    _cast_floats_tensors, apply_to_tensors)
```
Comment (baoleai) [inline torch_xla/experimental/spmd_fully_sharded_data_parallel.py]: apply_to_tensors is unused.

Comment (lausannel) [reply torch_xla/experimental/spmd_fully_sharded_data_parallel.py]: fixed

File: torch_xla/experimental/spmd_fully_sharded_data_parallel.py (hunk 1 above)
Comment (lausannel) [reply torch_xla/experimental/spmd_fully_sharded_data_parallel.py]: fixed

**Review by baoleai — APPROVED** [review]

**Pull request merged**